pkg/utils/test/webhook.go

```diff
@@ -50,24 +51,14 @@ func ValidationWebhookFailed(causeRegexes ...string) func(*testing.T, *admission
 		require.False(t, response.Allowed)
 
 		if len(causeRegexes) > 0 {
-			require.NotNil(t, response.Result.Details, "Response must include failure details")
+			require.NotNil(t, response.Result.Message, "Response must include failure message")
 		}
 
 		for _, cr := range causeRegexes {
-			found := false
 			t.Logf("Checking for existence of: %s", cr)
-			for _, cause := range response.Result.Details.Causes {
-				reason := fmt.Sprintf("%s: %s", cause.Field, cause.Message)
-				t.Logf("Reason: %s", reason)
-				match, err := regexp.MatchString(cr, reason)
-				require.NoError(t, err, "Match '%s' returned error: %v", cr, err)
-				if match {
-					found = true
-					break
-				}
-			}
-
-			require.True(t, found, "[%s] is not present in cause list", cr)
+			match, err := regexp.MatchString(cr, response.Result.Message)
+			require.NoError(t, err, "Match '%s' returned error: %v", cr, err)
+			require.True(t, match, "[%s] is not present in cause list", cr)
```

Comment:
I need to better understand why this behaviour changed. Obviously I switched these tests to use our webhook implementation (which I think is a good thing as we are using that implementation for the real webhooks as well) but a quick look a controller-runtime left me wondering where these details were populated. But I was in a rush ...